crypto/rand/rand_lib.c

```diff
@@ -638,6 +650,10 @@ int rand_pool_add(RAND_POOL *pool,
     }
 
     if (len > 0) {
+        if (pool->buffer + pool->len == buffer) {
```

Comment:
I think I need to add pool->alloc_len > pool->len to avoid comparing the address behind pool->buffer
which points to nothing and is not guaranteed to be different than buffer.
So this comparison which could be considered indeterminate.
FYI I would like to make sure that the result from rand_pool_add_begin is used with
rand_pool_add instead of rand_pool_add_end (these are easy to mix up, at least for me)

Reply:
rand_pool_add is the one-shot variant and rand_pool_add_{begin,end} create a group for adding entropy in several small steps. So they are never used at the same time. Why exactly do you find them confusing?

Reply:
I could drop this one, but found it confusing that rand_pool_add and rand_pool_add_end
could be confused and there is no protection against passing the buffer from
rand_pool_add_begin to rand_pool_add, which this would provide.

Reply:
also all parameters are identical...